src/gui/qgsmaptoolidentify.cpp

```diff
@@ -1176,11 +1176,17 @@ bool QgsMapToolIdentify::identifyRasterLayer( QList<IdentifyResult> *results, Qg
           // the return value should be interpreted as QMetaType::Type"
           if ( static_cast<QMetaType::Type>( value.userType() ) == QMetaType::Float )
           {
-            valueString = QgsRasterBlock::printValue( value.toFloat() );
+            const float fValue { value.toFloat() };
+            valueString = QgsRasterBlock::printValue( fValue );
```

Comment:
QgsRasterBlock::printValue isn't exposed to Python, and this is the ONLY use of it. I'd just move the locale handling into that function directly...

Reply:
Actually, I checked and there are multiple uses of that function, not all of them expect a localized string.

```text
grep -ril 'QgsRasterBlock::printValue' src|egrep -v 'sorti|prepare'
src/providers/grass/qgsgrassrasterprovider.cpp
src/core/raster/qgsrasterblock.cpp
src/core/raster/qgscontrastenhancement.cpp
src/core/raster/qgsrastertransparency.cpp
src/core/raster/qgssinglebandpseudocolorrenderer.cpp
src/core/raster/qgsrasterlayer.cpp
src/core/raster/qgscolorrampshader.cpp
src/gui/raster/qgsrastertransparencywidget.cpp
src/gui/raster/qgsrasterlayerproperties.cpp
src/gui/qgsrasterlayersaveasdialog.cpp
src/gui/qgsmaptoolidentify.cpp
```

Reply:
Argh sorry, I was looking at the float version only. Indeed the double version IS exposed to Python.

Could we add an optional `bool respectLocale = false` argument to printValue though?

Reply:
Done, test added.